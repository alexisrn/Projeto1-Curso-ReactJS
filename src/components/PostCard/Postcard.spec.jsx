const { render, screen } = require("@testing-library/react");
const { PostCard } = require(".");
const { PostCardMock } = require("./mock");

const props = PostCardMock;

describe("<PostCard />", () => {
  it("should  render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', {name: 'title'})).toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', {name: 'title'})).toBeInTheDocument();

    expect(screen.getByText('body')).toBeInTheDocument();
    
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  })
});
