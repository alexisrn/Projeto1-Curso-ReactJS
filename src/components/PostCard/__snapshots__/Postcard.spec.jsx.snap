// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      title
    </h2>
    <p>
      body
    </p>
  </div>
</div>
`;
